wgsl/index.bs

```diff
@@ -3621,7 +3621,15 @@ In this case the call site of this function invocation evaluates to the return v
 The type of the return value must match the return type of the function.
 
 
-### Discard TODO ### {#discard-statement}
+### Discard ### {#discard-statement}
+
+The `discard` statement causes the current invocation to terminate and no
+entry point return value to be generated for the invocation. Only statements
+executed prior to the `discard` statement will have observable side effects. A
+`discard` statement may be executed by any function and the effect is the same,
+termination of the invocation occurs immediately upon executing the statement.
+After a `discard` statement is executed, control flow is non-uniform for the
```

Comment:
We may need to talk about this. Let's consider a case like this:
```rust
if (false) { discard; }
```
Technically it's never executed. Should we still consider the control flow to be non-uniform?

Or, more generally,
```rust
if (uniform_expression) {
  textureSample() // something that requires control flow being uniform
} else {
  discard; // makes the control flow non-uniform
}
```

If I understand correctly, this condition, as well as everything after it, is still in uniform control flow: either branch is taken by all invocations, and either all of them are discarded (in which case nothing past the condition matters anyway), or none of them.

Reply:
For the first example, I phrased the spec such that the lose of uniformity only occurs if the discard is executed which I don't is the case.

For the second example, that will depend on the uniformity analysis and whether it counts the uniform expression as uniform or not.